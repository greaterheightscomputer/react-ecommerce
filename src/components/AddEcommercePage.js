import React from 'react';

const AddEcommercePage = () => (
    <div>
        This is from my add expense component
    </div>
);

export default AddEcommercePage;