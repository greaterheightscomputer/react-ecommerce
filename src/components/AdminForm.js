import React from 'react';
import { SingleDatePicker } from 'react-dates';
import moment from 'moment';

export default class AdminForm extends React.Component{    
    constructor(props) { //its will be use for both AddProduct and EditProduct
        super(props);

        this.state = {
            description: props.ecommerce ? props.ecommerce.description : '',
            category: props.ecommerce ? props.ecommerce.category : '',
            item: props.ecommerce ? props.ecommerce.item : '',
            amount: props.ecommerce ? (props.ecommerce.amount / 100).toString() : '',
            stock: props.ecommerce ? props.ecommerce.stock :'',       
            image: props.ecommerce ? props.ecommerce.image.name : '',
            imageUrl: props.ecommerce ? props.ecommerce.imageUrl : '',
            company: props.ecommerce ? props.ecommerce.company : '',
            info: props.ecommerce ? props.ecommerce.info: '',
            createdAt: props.ecommerce ? moment(props.ecommerce.createdAt) : moment(),
            focused: false,
            error: ''               
        }
    };
    onDescriptionChange = (e) => {
        const description = e.target.value;
        this.setState(() => ({ description }));
    };
    onCategoryChange = (e) => {
        const category = e.target.value;
        this.setState(() => ({ category }));
    };
    onItemChange = (e) => {
        const item = e.target.value;
        this.setState(() => ({ item }));
    };
    onStockChange = (e) => {
        const stock = e.target.value;        
        this.setState(() => ({ stock }));
    };
    onAmountChange = (e) => {
        const amount = e.target.value;
        if (!amount || amount.match(/^\d{1,}(\.\d{0,2})?$/)) {
            this.setState(() => ({ amount }));        
        }        
    };
    onCompanyChange = (e) => {
        const company = e.target.value;
        this.setState(() => ({ company }));
    };
    onInfoChange = (e) => {
        const info = e.target.value;
        this.setState(() => ({ info }));
    };
    onDateChange = (createdAt) => {
        if (createdAt) {
            this.setState(() => ({ createdAt }));            
        }        
    };
    onFocusChange = ({ focused}) => {
        this.setState(() => ({ focused }));
    };
    onImageChange = (e) => {        
        let reader = new FileReader();
        let file = e.target.files[0];        
        reader.onloadend = () => {
            this.setState({
                image: file,
                imageUrl: reader.result
            });
        };
        reader.readAsDataURL(file); 
    };
    onSubmitClick = (e) => {
        e.preventDefault();
        if (!this.state.description || !this.state.category || !this.state.item || !this.state.amount || !this.state.image || !this.state.stock || !this.state.company || !this.state.info) {        
            this.setState(() => ({
                error: 'Please provide value for the empty field(s)'
            }));    
        } else {
            this.setState(() => ({ error: '' }));            
            this.props.onSubmit({
                description: this.state.description,
                category: this.state.category,
                item: this.state.item,
                amount: parseFloat(this.state.amount, 10) * 100,
                stock: this.state.stock,
                image: this.state.image.name,
                imageUrl: this.state.imageUrl,
                company: this.state.company,
                info: this.state.info,
                createdAt: this.state.createdAt.valueOf()
            });            
        } 
    };
    render() {
        return (
            <form className="form" onSubmit={this.onSubmitClick}>                
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Description"
                        autoFocus
                        value={this.state.description}
                        onChange={this.onDescriptionChange}
                    />
                    <select
                        className= "select"
                        value={this.state.category}
                        onChange={this.onCategoryChange}                        
                    >
                        <option value="">Select Category</option>
                        <option value="boy">Boy</option>
                        <option value="girl">Girl</option>
                        <option value="amen">Men</option>
                        <option value="women">Women</option>
                    </select>      
                    <select
                        className= "select"
                        value={this.state.item}
                        onChange={this.onItemChange}>
                        <option value="">Select Item</option>
                        {/*boyItems */}
                        <option value="boy_bag">Boy Bag</option>
                        <option value="boy_belt">Boy Belt</option>
                        <option value="boy_clothe">Boy Clothe</option>
                        <option value="boy_footwear">Boy Footwear</option>
                        <option value="boy_headwear">Boy Headwear</option>
                        <option value="boy_jewellery">Boy Jewellery</option>
                        <option value="boy_tie">Boy Tie</option>
                        <option value="boy_underwear">Boy Underwear</option>
                        {/*girlItems */}
                        <option value="girl_bag">Girl Bag</option>
                        <option value="girl_belt">Girl Belt</option>
                        <option value="girl_clothe">Girl Clothe</option>
                        <option value="girl_footwear">Girl Footwear</option>
                        <option value="girl_headwear">Girl Headwear</option>
                        <option value="girl_jewellery">Girl Jewellery</option>
                        <option value="girl_tie">Girl Tie</option>
                        <option value="girl_underwear">Girl Underwear</option>
                        {/*menItems */}
                        <option value="men_bag">Men Bag</option>
                        <option value="men_belt">Men Belt</option>
                        <option value="men_clothe">Men Clothe</option>
                        <option value="men_footwear">Men Footwear</option>
                        <option value="men_headwear">Men Headwear</option>
                        <option value="men_jewellery">Men Jewellery</option>
                        <option value="men_tie">Men Tie</option>
                        <option value="men_underwear">Men Underwear</option>
                        {/*womenItems */}
                        <option value="women_bag">Women Bag</option>
                        <option value="women_belt">Women Belt</option>
                        <option value="women_clothe">Women Clothe</option>
                        <option value="women_footwear">Women Footwear</option>
                        <option value="women_headwear">Women Headwear</option>
                        <option value="women_jewellery">Women Jewellery</option>
                        <option value="women_tie">Women Tie</option>
                        <option value="women_underwear">Women Underwear</option>
                    </select>    
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Amount"                        
                        value={this.state.amount}
                        onChange={this.onAmountChange}
                    />
                    <input
                        className= "text-input"
                        type="number"
                        placeholder="Stock"                        
                        value={this.state.stock}
                        onChange={this.onStockChange}
                    />
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Company Name"                        
                        value={this.state.company}
                        onChange={this.onCompanyChange}
                    />
                    <input
                        className= "text-input"
                        type="text"
                        placeholder="Product Information"                        
                        value={this.state.info}
                        onChange={this.onInfoChange}
                    />
                    <SingleDatePicker
                        date={this.state.createdAt}
                        onDateChange={this.onDateChange}
                        focused={this.state.focused}
                        onFocusChange={this.onFocusChange}
                        numberOfMonths={1}
                        isOutsideRange={() => false}
                    />
                   <div>
                        <input 
                            className= "text-input"
                            type="file"                            
                            onChange={this.onImageChange}
                        />
                        <div className="image-input">                            
                            <img src={this.state.imageUrl} height="200" width="200"/>                            
                        </div>                        
                    </div>
                <div>
                    {this.state.error && <p className="form__error">{this.state.error}</p>}
                </div>
                <div>
                    <button className="button">Save Product</button>                         
                </div>                   
            </form>                            
        );
    };
}

