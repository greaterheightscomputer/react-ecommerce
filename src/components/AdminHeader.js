import React from 'react';
import { Link } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const AdminHeader = ({ startLogout }) => (
    <header className="admin-header">        
        <div className="content-container">
            <div className="admin-header__container">
                <Link className="admin-header__title " to="/admin_dashboard">
                    <h1>E-Market</h1>
                </Link>
                <button className="button button--link" onClick={startLogout}>Logout</button>          
            </div>            
        </div>        
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(AdminHeader);