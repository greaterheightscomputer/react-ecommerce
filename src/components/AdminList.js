import React from 'react';
import { connect } from 'react-redux';
import AdminListItem from './AdminListItem';
import selectEcommerce from '../selectors/ecommerces';

export const AdminList = (props) => (
    
    <div className="content-container">
        <div className="list-header">
            <div>Admin Products List</div>
        </div>             
        <div className="list-body">
            <div className="list-items" >
                {
                    props.ecommerces.length === 0 ? (
                        <div className="list-item list-item--message">
                            <span>No Products</span>
                        </div>
                    ) : (
                        props.ecommerces.map((ecommerce) =>{
                            return <AdminListItem key={ecommerce.id} {...ecommerce} />
                        })
                    )
                }                        
            </div>        
        </div>           
    </div>
);

const mapStateToProps = (state) => {
    return {
        ecommerces: selectEcommerce(state.ecommerces, state.filters)        
    };
}

export default connect(mapStateToProps)(AdminList);