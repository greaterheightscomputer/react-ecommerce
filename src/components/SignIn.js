import React from 'react';
import { connect } from 'react-redux';
import { startLogin } from '../actions/auth';

export class SignIn extends React.Component {
    state = {
        email: '',
        password: ''        
    };    
    onEmailChange = (e) => {      
        const email = e.target.value  
        this.setState({ email });
    };
    onPasswordChange = (e) => {      
        const password = e.target.value  
        this.setState({ password });
    };
    onSubmitClick = (e) => {
        e.preventDefault();                  
        //console.log(this.state);
        this.props.startLogin(this.state);          
    };
    render() {
        const { authError } = this.props;
        return (
            <div className="box-layout">                
                <div className="box-layout__box">
                    <form onSubmit={this.onSubmitClick} >
                        <h1 className="box-layout__title"> Sign In</h1>
                        <div>
        {/*<label>Email</label>*/}
                            <input 
                                className="text-input text-input__extend"
                                type="email"                             
                                autoFocus
                                value={this.state.email} 
                                onChange={this.onEmailChange}
                                placeholder="Enter Email"
                            />
                        </div>
                        <div>
        {/*<label>Password</label>*/}
                            <input
                                className="text-input text-input__extend"
                                type="password"                             
                                value={this.state.password} 
                                onChange={this.onPasswordChange} 
                                placeholder="Enter Password"
                            />
                        </div>
                        <div>
                            <button className="button button__space">Login with Credential</button>
                            <div className="box-layout__error">
                                { authError ? <p>{authError}</p> : null }
                            </div>
                        </div>
                    </form>
                </div>                
            </div>
        );
    };
};

const mapStateToProps = (state) => ({
    authError: state.auth.authError
});

const mapDispatchToProps = (dispatch) => ({
    startLogin: (creds) => dispatch(startLogin(creds))
});

export default connect(mapStateToProps, mapDispatchToProps)(SignIn);