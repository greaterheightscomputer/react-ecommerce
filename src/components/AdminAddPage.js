import React from 'react';
import { connect } from 'react-redux';
import AdminForm from './AdminForm';
import { startAddEcommerce } from '../actions/ecommerces';

export class AdminAddPage extends React.Component {
    onSubmit = (ecommerce) => {
        // console.log(ecommerce)
        // props.dispatch(addEcommerce(ecommerce));
        this.props.startAddEcommerce(ecommerce);
        this.props.history.push('/admin_dashboard');
    };
    render() {
        return (
            <div className="list-body">
                <div className="page-admin-header">
                    <div className="content-container">
                        <h1 className="page-admin-header__title"> Add Product</h1>
                    </div>
                </div>                
                <div className="content-container">
                    <AdminForm
                        onSubmit={this.onSubmit}
                    />
                </div>
            </div>
        );
    };
};

const mapDispatchToProps = (dispatch) => ({
    startAddEcommerce: (ecommerce) => dispatch(startAddEcommerce(ecommerce))
});

export default connect(undefined, mapDispatchToProps)(AdminAddPage);