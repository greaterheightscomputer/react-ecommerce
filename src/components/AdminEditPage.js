import React from 'react';
import { connect } from 'react-redux';
import AdminForm from './AdminForm';
import { startEditEcommerce, startRemoveEcommerce } from '../actions/ecommerces';

export class AdminEditPage extends React.Component {
    onSubmit = (ecommerce) => {
        // console.log('updated', ecommerce);
        this.props.startEditEcommerce(this.props.ecommerce.id, ecommerce);
        this.props.history.push('/admin_dashboard');        
    };
    onRemove = () => {
        this.props.startRemoveEcommerce({ id: this.props.ecommerce.id });            
        this.props.history.push('/admin_dashboard');
    };
    render(){
        return(
            <div>
                <div className="page-admin-header">
                    <div className="content-container">
                        <h1 className="page-admin-header__title">Edit Product</h1>
                    </div>
                </div>
                <div className="content-container">
                    <AdminForm 
                    ecommerce={this.props.ecommerce}
                    onSubmit={this.onSubmit}
                    />
                    <button className="button button--secondary" onClick={this.onRemove}>Remove Product</button>
                </div>
            </div>
        );        
    };
};

const mapStateToProps = (state, props) => ({
    ecommerce: state.ecommerces.find((ecommerce) => ecommerce.id === props.match.params.id)
});

const mapDispatchToProps = (dispatch, props) => ({
    startEditEcommerce: (id, ecommerce) => dispatch(startEditEcommerce(id, ecommerce)),
    startRemoveEcommerce: (data) => dispatch(startRemoveEcommerce(data)) 
});

export default connect(mapStateToProps, mapDispatchToProps)(AdminEditPage);