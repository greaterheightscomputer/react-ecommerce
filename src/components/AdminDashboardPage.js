import React from 'react';
import AdminList from './AdminList';
import AdminListFilters from './AdminListFilters';
import AdminSummary from './AdminSummary';

const AdminDashboardPage = () => (
    <div>        
        <AdminSummary />
        <AdminListFilters />
        <AdminList />        
    </div>
);

export default AdminDashboardPage;