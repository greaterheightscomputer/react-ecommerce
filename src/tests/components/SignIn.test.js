import React from 'react';
import { shallow } from 'enzyme';
import { SignIn } from '../../components/SignIn';

test('should render SignIn with default data correctly', () => {
    const wrapper = shallow(<SignIn />);
    expect(wrapper).toMatchSnapshot();
});

test('should set email on input change', () => {
    const value = 'adebarakhadijat@gmail.com';
    const wrapper = shallow(<SignIn />);
    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('email')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should set password on input change', () => {
    const value = '123abc';
    const wrapper = shallow(<SignIn />);
    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('password')).toBe(value);
    expect(wrapper).toMatchSnapshot();
});

test('should call props.onSbmit for valid form submission', () => {
    const onSubmitSpy = jest.fn();        
    const wrapper = shallow(<SignIn startLogin={onSubmitSpy} />);
    wrapper.find('form').simulate('submit', {
        preventDefault: () => { }
    });
    onSubmitSpy({
        email: 'adebarakhadijat@gmail.com',
        password: '123abc'        
    });            
    expect(onSubmitSpy).toHaveBeenLastCalledWith({
        email: 'adebarakhadijat@gmail.com',
        password: '123abc'
    }); 
});

