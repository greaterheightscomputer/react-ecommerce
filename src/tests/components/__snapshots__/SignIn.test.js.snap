// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render SignIn with default data correctly 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <form
      onSubmit={[Function]}
    >
      <h1
        className="box-layout__title"
      >
         Sign In
      </h1>
      <div>
        <input
          autoFocus={true}
          className="text-input text-input__extend"
          onChange={[Function]}
          placeholder="Enter Email"
          type="email"
          value=""
        />
      </div>
      <div>
        <input
          className="text-input text-input__extend"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
      </div>
      <div>
        <button
          className="button button__space"
        >
          Login with Credential
        </button>
        <div
          className="box-layout__error"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`should set email on input change 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <form
      onSubmit={[Function]}
    >
      <h1
        className="box-layout__title"
      >
         Sign In
      </h1>
      <div>
        <input
          autoFocus={true}
          className="text-input text-input__extend"
          onChange={[Function]}
          placeholder="Enter Email"
          type="email"
          value="adebarakhadijat@gmail.com"
        />
      </div>
      <div>
        <input
          className="text-input text-input__extend"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value=""
        />
      </div>
      <div>
        <button
          className="button button__space"
        >
          Login with Credential
        </button>
        <div
          className="box-layout__error"
        />
      </div>
    </form>
  </div>
</div>
`;

exports[`should set password on input change 1`] = `
<div
  className="box-layout"
>
  <div
    className="box-layout__box"
  >
    <form
      onSubmit={[Function]}
    >
      <h1
        className="box-layout__title"
      >
         Sign In
      </h1>
      <div>
        <input
          autoFocus={true}
          className="text-input text-input__extend"
          onChange={[Function]}
          placeholder="Enter Email"
          type="email"
          value=""
        />
      </div>
      <div>
        <input
          className="text-input text-input__extend"
          onChange={[Function]}
          placeholder="Enter Password"
          type="password"
          value="123abc"
        />
      </div>
      <div>
        <button
          className="button button__space"
        >
          Login with Credential
        </button>
        <div
          className="box-layout__error"
        />
      </div>
    </form>
  </div>
</div>
`;
